site/src/routes/nfts-get.js

```diff
@@ -16,14 +17,27 @@ export const status = async (event, params) => {
   const auth = await validate(event)
   const user = auth.user
 
-  const [nft, deals] = await Promise.all([
+  const [nft, pin, deals] = await Promise.all([
     getNft({ user, cid: params.cid }),
+    getPin(params.cid),
     getDeals(params.cid),
   ])
 
   if (nft == null) {
-    return HTTPError.respond(new HTTPError('not found', 404))
+    return HTTPError.respond(new HTTPError('NFT not found', 404))
   }
 
-  return new JSONResponse({ ok: true, value: { ...nft, deals } })
+  if (pin == null) {
+    return HTTPError.respond(new HTTPError('pin not found', 404))
+  }
+
+  /** @type {import('../bindings').NFTResponse} */
+  const res = {
+    ...nft,
+    size: pin.size,
+    pin: { cid: nft.cid, ...(nft.pin || {}), ...pin },
```

Comment:
can't we simplify this pin structure everywhere ? avoid  `{ cid: nft.cid, ...(nft.pin || {}), ...pin }` and just have everything in the output from getPin ?

Reply:
We could include the `cid` but `nft.pin` is user provided pin `name` and `meta` so we'd still have to merge it here.

Reply:
I will add the `cid` because that will help a little!